### PR TITLE
feat(agents): persist agent capabilities

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- bump agynio/api buf dependency and regenerate bindings
- add agent capabilities column and store serialization
- map agent capabilities through create/get/list/update requests

## Testing
- go test ./...
- go build ./...

Fixes #133

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Test & Lint Summary
Commands:
- go test ./...
- go build ./...

Tests: Passed: 1, Failed: 0, Skipped: 0.
Lint: not configured (not run).

File: internal/store/store.go [hunk 1]
```diff
@@ -48,9 +49,35 @@ func stringPtrFromPg(value pgtype.Text) *string {
 	return &text
 }
 
+func decodeCapabilities(value []byte) ([]string, error) {
+	if value == nil {
+		return nil, fmt.Errorf("capabilities is NULL")
+	}
+	var capabilities []string
```
Comment (noa-lucent) [inline internal/store/store.go:56]: [minor] `decodeCapabilities` and `encodeCapabilities` introduce JSON serialization logic that warrants unit tests — especially the edge cases:

- `decodeCapabilities(nil)` → error (NULL invariant)
- `decodeCapabilities([]byte("null"))` → error (JSON null invariant)
- `decodeCapabilities([]byte("[]"))` → empty `[]string{}`
- `decodeCapabilities([]byte(`["a","b"]`))` → `["a","b"]`
- `encodeCapabilities(nil)` → `[]byte("[]")` (nil normalization)
- `encodeCapabilities([]string{})` → `[]byte("[]")`

These are easy to test and document the contract for future readers.

**Review by noa-lucent — APPROVED** [review]
Clean implementation that follows existing codebase patterns well. Key observations:

- **Migration**: Correct — `JSONB NOT NULL DEFAULT '[]'` handles existing rows properly.
- **Store layer**: `encode`/`decodeCapabilities` are well-structured. The nil-to-empty normalization in `encodeCapabilities` correctly prevents writing SQL/JSON `null` to a NOT NULL column. The `decodeCapabilities` invariant checks catch DB-level corruption.
- **Server layer**: Proto3 repeated field presence detection via `req.Capabilities != nil` is the correct approach, and the explanatory comment is appreciated.
- **Converter**: Defensive `append([]string(nil), ...)` copies at the boundary are good practice.
- **SQL**: Column ordering, scan ordering, and parameter counts all verified correct.

One minor suggestion for unit test coverage on the encode/decode helpers. LGTM overall.
